ext/drmgr/drmgr.h

```diff
@@ -711,6 +711,58 @@ enum {
     DRMGR_NOTE_NONE = 0, /* == DRX_NOTE_NONE */
 };
 
+/**
+ * TODO: Full doxygen comments will be done as part of upstream pull request,
+ *       when code has been reviewed.
+ */
+enum {
+    DRMGR_NOTE_EMUL_START,
+    DRMGR_NOTE_EMUL_STOP,
+    DRMGR_NOTE_EMUL_COUNT,
+};
+
+/**
+ * TODO: Full doxygen comments will be done as part of upstream pull request,
+ *       when code has been reviewed.
+ */
+enum {
+    DRMGR_EMUL_INSTR_PC,
+    DRMGR_EMUL_INSTR,
+    DRMGR_EMUL_ISA_VERSION,
+};
+
+/**
+ * TODO: Full doxygen comments will be done as part of upstream pull request,
+ *       when code has been reviewed.
+ */
+DR_EXPORT
+void *
+set_emul_note_val(int enote_val);
```

Comment:
I think we would want a namespace prefix on the function names, sthg like `drmgr_emul_`.

WDYT of instead having an interface like this where drmgr hides some of these details with a slightly higher abstraction level:

```
drmgr_create_emulation_start(app_pc orig_pc, instr_t *orig_instr, int isa_version)
drmgr_create_emulation_end()
```

These create the label instruction and populate its note and data fields.  Then we have:
```
bool drmgr_is_emulation_start(app_pc *orig_pc, instr_t **orig_instr, int *isa_version)
bool drmgr_is_emulation_stop()
```

To support adding new fields maybe the start ones should take in a struct with a size field instead of inlined parameters.

Reply:
I think this is a good idea. I've implemented it as best I can and it simplifies clients as well as having a consistent naming convention.